src/pybind/mgr/dashboard/HACKING.rst

```diff
@@ -2312,3 +2312,104 @@ plugin could be rewritten like this:
+Dashboard.
+
+The development environment for Ceph Dashboard has live reloading enabled so
+any changes made in UI are reflected in open browser windows. Ceph Dashboard
+uses Bootstrap as the main third-party CSS library.
+
+Avoid duplication of code. Be consistent with the existing UI by reusing
+existing SCSS declarations as much as possible.
+
+Colors
+~~~~~~
+All the colors used in Ceph Dashboard UI are listed in
+`frontend/src/styles/defaults/_bootstrap-defaults.scss`. If using new color
+always define color variables in the `_bootstrap-defaults.scss` and
+use the variable instead of hard coded color values so that changes to color
+are reflected in similar UI elements.
+
+The main color for the Ceph Dashboard is `$primary`. The primary color is
+used in navigation components and as the `border-color` for input components of
+form.
+
+The secondary color is `$secondary` and is the background color for Ceph
+Dashboard.
+
+Buttons
+~~~~~~~
+Buttons are used for performing actions such as: “Submit”, “Edit, “Create" and
+“Update”. These actions almost always occur on the same page.
+
+**Forms:** When using to submit forms anywhere in the dashboard, the main action
+button should of contrasting color and the secondary button should be of white
+color. The text on the action button should be same as the form title and follow
+a title case. The text on the secondary button should be `Cancel`. Perform the
+action button should always be on right while `Cancel` button should always be
+on left.
+
+**Modals**: The main button should be of the same color as the main action
+button and the secondary action button should be white. The text on the action
+button should follow a title case and correspond to the action to be performed.
+The text on the secondary button should be `Close`.
+
+**Disclosure Button:** Disclosure buttons should be used to allow users to
+display and hide additional content in the interface.
+
+**Action Button**: Use action button to perform actions such as edit or update a
+component. The button color should be same as the form's main button color. The
+text on button should follow title case and have an icon corresponding to the
+actions they perform.
+
+**Drop Down Buttons:** Use dropdown buttons to display predefined lists of
+actions. All drop down buttons have icons corresponding to the action they
+perform.  
+
+Links
+~~~~~
+Use text hyperlinks as navigation to guide users to a new page in the application
+or to anchor users to a section within a page. The color of the hyperlinks
+should be `$primary`.
+
+Tabs
+~~~~
+The current opened tab should have black font color and the link to other tabs
+should have `$primary` font.
+
+Forms
+~~~~~
+Mark invalid form fields with red outline and show a meaningful error message.
+Use red as font color for message and be as specific as possible.
+Mark valid forms with a green outline and a green tick at the end of the form.
```

Comment:
The colors are defined automatically and the developer doesn't need to worry about it.
Would suggest mentioning the following:

- I agree with asking for a meaningful error message.
- We should use `This field is required.` as the exact error message for required fields.
- No form control should modify the content above it, it should always be bellow.
- We should not have sections with a bigger header than the parent.